server/auth/google.js

```diff
@@ -43,6 +44,13 @@ router.get('google.callback', async ctx => {
     return;
   }
 
+  // allow all domains by default if the env is not set
+  const allowedDomains = allowedDomainsEnv && allowedDomainsEnv.split(',');
+  if (allowedDomains && allowedDomains.indexOf(profile.data.hd) < 0) {
```

Comment:
You can use `allowedDomains.includes(profile.data.hd)` as this is server side.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes